doc: update README to align with the new project brand (#158)
notaryproject/notation-core-go: README.md

> # Notation-core-go library

Reviewer: Seems a different pattern from the `notation-go` readme. Maybe we can try `Notation Core Golang Library`

> # Notation-core-go library

Reviewer 2: Why are we making this change? I think we can keep the title to be same as the repo's name, i.e. `notation-core-go`.

[![Build Status](https://github.com/notaryproject/notation-core-go/actions/workflows/build.yml/badge.svg?event=push&branch=main)](https://github.com/notaryproject/notation-core-go/actions/workflows/build.yml?query=workflow%3Abuild+event%3Apush+branch%3Amain)
[![codecov](https://codecov.io/gh/notaryproject/notation-core-go/branch/main/graph/badge.svg)](https://codecov.io/gh/notaryproject/notation-core-go)
[![Go Reference](https://pkg.go.dev/badge/github.com/notaryproject/notation-core-go.svg)](https://pkg.go.dev/github.com/notaryproject/notation-core-go@main)

> Notation-core-go library provides specific implementation for Notary Project signature envelope, signature format, and OCSP revocation check.

Reviewer 2: nit:
```suggestion
notation-core-go library provides specific implementation for Notary Project signature envelope, signature format, and OCSP revocation check.
```

> Notation-core-go library provides specific implementation for Notary Project signature envelope, signature format, and OCSP revocation check.

Reviewer 3: ```suggestion
notation-core-go implements signature generation, parsing and revocation related functionalities based on [Notary Project specifications](link).
```

> Notation-core-go library reaches stable in July 2023 and is in active development status.

Reviewer 2: ```suggestion
notation-core-go library reaches stable in July 2023 and is in active development status.
```

> > **Note** The Notation documentation is available [here](https://notaryproject.dev/docs/). You can also find the Notary Project [README](https://github.com/notaryproject/.github/blob/main/README.md) to learn about the overall Notary Project.

Reviewer: Suggest moving overall intro to line 10

## Table of Contents
- [Core Documents](#core-documents)
- [Code of Conduct](#code-of-conduct)
- [License](#license)
 

> ## Core Documents

Reviewer 3: ```suggestion
## Documentation
```

All library documentation is available at [Go Reference](https://pkg.go.dev/github.com/notaryproject/notation-core-go).

> ## Code of Conduct

Reviewer: Suggest referencing to `https://github.com/notaryproject/.github/blob/main/CODE_OF_CONDUCT.md`, which was updated months ago per TOC review.

[![codecov](https://codecov.io/gh/notaryproject/notation-core-go/branch/main/graph/badge.svg)](https://codecov.io/gh/notaryproject/notation-core-go)
[![Go Reference](https://pkg.go.dev/badge/github.com/notaryproject/notation-core-go.svg)](https://pkg.go.dev/github.com/notaryproject/notation-core-go@main)

> notation-core-go implements signature generation, parsing and revocation related functionalities based on the [Notary Project specifications](https://github.com/notaryproject/specifications).

Reviewer 2: nit:
notation-core-go also provides `certificate and certificate chain validation`.

> notation-core-go implements signature generation, parsing and revocation related functionalities based on the [Notary Project specifications](https://github.com/notaryproject/specifications).

Reviewer 4: Should emphasize on that `notation-core-go` provides core crypto functionality for `notation-go`.

> notation-core-go reached a stable release as of July 2023 and continues to be actively developed.

Reviewer 4: developed **and maintained**?

You can also find the Notary Project [README](https://github.com/notaryproject/.github/blob/main/README.md) to learn about the overall Notary Project.

> > **Note** The Notary Project documentation is available [here](https://notaryproject.dev/docs/).

Reviewer 4: ```suggestion
> [!NOTE]
> The Notary Project documentation is available [here](https://notaryproject.dev/docs/).
```
> [!NOTE]
> The Notary Project documentation is available [here](https://notaryproject.dev/docs/).

## Table of Contents

> - [Core Documents](#core-documents)

Reviewer 3: ```suggestion
- [Documentation](#documentation)
```
